### PR TITLE
Fixes runtime issues in code.

### LINKED ISSUE #5365: unsorted.dm runtime
## Testmerges:

#5285 #5241 #5159 #5217 #5242 #5253

## Runtime:

Runtime Viewer

<<<Refresh

[02:58:48] Runtime in unsorted.dm, line 35: addtimer called with a 0 or negative wait. Converting to 0.5
proc name: stack trace (/proc/stack_trace)
usr: Overlord1234/(Overlord1234)
usr.loc: (start area (8, 8, 1))
src: null
call stack:
stack trace("addtimer called with a 0 or ne...")
addtimer(/datum/callback (/datum/callback), 0, 0, "code/modules/client/client_pro...", 769)
Overlord1234 (/client): change view("15x15")
Overlord1234 (/mob/new_player): Login()
Overlord1234 (/mob/new_player): Login()

usr: VV PP FLW
usr.loc: VV JMP

### LINKED ISSUE #5376: More 0 len timers
## Testmerges:

Round ID: [6181](https://scrubby.melonmesa.com/round/6181)
BYOND Version: 513.1534
Server revision compiled on: 
Master commit: 3a28a4ec15d891ca8ad610739a15219941070e5e
The following pull requests are currently test merged:
#5285: 'some module changes to tyr/mimir mark 1' by Pariah919 at commit 6bec2c452f
#5241: 'Adds mining well upgrades(Ready)' by MLGTASTICa at commit 2d868ed568
#5159: 'QoL Improvements for Hivelord Tunnels' by Surrealaser at commit 349eb7f1d7
#5217: 'Drone's Salvage Plasma Reworked into Salvage Biomass' by Surrealaser at commit 74d3442f2c
#5242: 'Reworks IFF code, Gives Aim Mode to select rifles' by Walarks at commit 259ba97f3d
#5253: 'Stuns changes part 2 - defender' by Pariah919 at commit 000ea3a015

## Reproduction:

proc name: stack trace (/proc/stack_trace)
usr: XTrihard747/(David Dick)
usr.loc: (Aerodrome Landing Pad (161, 88, 2))
src: null
call stack:
stack trace("addtimer called with a 0 or ne...")
addtimer(/datum/callback (/datum/callback), 0, 0, "code/modules/client/client_pro...", 769)
XTrihard747 (/client): change view("15x15")
the T-42 light machine gun (Wi... (/obj/item/weapon/gun/rifle/standard_lmg): zoom(David Dick (/mob/living/carbon/human), 11, 12)
the T-42 light machine gun (Wi... (/obj/item/weapon/gun/rifle/standard_lmg): zoom(David Dick (/mob/living/carbon/human), 11, 12)
David Dick (/mob/living/carbon/human): Moved(the plating (161,87,2) (/turf/open/floor/plating/icefloor), 1)
David Dick (/mob/living/carbon/human): Moved(the plating (161,87,2) (/turf/open/floor/plating/icefloor), 1)
David Dick (/mob/living/carbon/human): Moved(the plating (161,87,2) (/turf/open/floor/plating/icefloor), 1)
David Dick (/mob/living/carbon/human): Move(the plating (161,88,2) (/turf/open/floor/plating/icefloor), 1, null)
David Dick (/mob/living/carbon/human): Move(the plating (161,88,2) (/turf/open/floor/plating/icefloor), 1)
David Dick (/mob/living/carbon/human): Move(the plating (161,88,2) (/turf/open/floor/plating/icefloor), 1)
David Dick (/mob/living/carbon/human): Move(the plating (161,88,2) (/turf/open/floor/plating/icefloor), 1)
David Dick (/mob/living/carbon/human): Move(the plating (162,88,2) (/turf/open/floor/plating/icefloor), 5, null)
David Dick (/mob/living/carbon/human): Move(the plating (162,88,2) (/turf/open/floor/plating/icefloor), 5)
David Dick (/mob/living/carbon/human): Move(the plating (162,88,2) (/turf/open/floor/plating/icefloor), 5)
David Dick (/mob/living/carbon/human): Move(the plating (162,88,2) (/turf/open/floor/plating/icefloor), 5)
Ravage (200) (/datum/action/xeno_action/activable/ravage): use ability(the T-42 drum magazine (10x24m... (/obj/item/ammo_magazine/standard_lmg))
Mature Ravager (Lustius) (/mob/living/carbon/xenomorph/ravager): MiddleClickOn(the T-42 drum magazine (10x24m... (/obj/item/ammo_magazine/standard_lmg))
Mature Ravager (Lustius) (/mob/living/carbon/xenomorph/ravager): ClickOn(the T-42 drum magazine (10x24m... (/obj/item/ammo_magazine/standard_lmg), the plating (164,89,2) (/turf/open/floor/plating/icefloor), "icon-x=22;icon-y=11;middle=1;s...")
the T-42 drum magazine (10x24m... (/obj/item/ammo_magazine/standard_lmg): Click(the plating (164,89,2) (/turf/open/floor/plating/icefloor), "mapwindow.map", "icon-x=22;icon-y=11;middle=1;s...")
Doc Robert (/client): Click(the T-42 drum magazine (10x24m... (/obj/item/ammo_magazine/standard_lmg), the plating (164,89,2) (/turf/open/floor/plating/icefloor), "mapwindow.map", "icon-x=22;icon-y=11;middle=1;s...")

### LINKED ISSUE #5377: 0 len login timer
[17:20:12] Runtime in unsorted.dm, line 35: addtimer called with a 0 or negative wait. Converting to 0.5
proc name: stack trace (/proc/stack_trace)
usr: Eeskektisica/(Eeskektisica)
usr.loc: (start area (8, 8, 1))
src: null
call stack:
stack trace("addtimer called with a 0 or ne...")
addtimer(/datum/callback (/datum/callback), 0, 0, "code/modules/client/client_pro...", 769)
Eeskektisica (/client): change view("15x15")
Eeskektisica (/mob/new_player): Login()
Eeskektisica (/mob/new_player): Login()

### LINKED ISSUE #5379: Weeds 0 len timer
[19:49:23] Runtime in unsorted.dm, line 35: addtimer called with a 0 or negative wait. Converting to 0.5
proc name: stack trace (/proc/stack_trace)
src: null
call stack:
stack trace("addtimer called with a 0 or ne...")
addtimer(/datum/callback (/datum/callback), 0, 0, "code/controllers/subsystem/wee...", 37)
Weed Decay (/datum/controller/subsystem/weeds_decay): fire(0)
Weed Decay (/datum/controller/subsystem/weeds_decay): ignite(0)
Master (/datum/controller/master): RunQueue()
Master (/datum/controller/master): Loop()
Master (/datum/controller/master): StartProcessing(0)

### LINKED ISSUE #5380: Rocket pod runtime
## Testmerges:
Round ID: [6181](https://scrubby.melonmesa.com/round/6181)
BYOND Version: 513.1534
Server revision compiled on: 
Master commit: 3a28a4ec15d891ca8ad610739a15219941070e5e
The following pull requests are currently test merged:
#5285: 'some module changes to tyr/mimir mark 1' by Pariah919 at commit 6bec2c452f
#5241: 'Adds mining well upgrades(Ready)' by MLGTASTICa at commit 2d868ed568
#5159: 'QoL Improvements for Hivelord Tunnels' by Surrealaser at commit 349eb7f1d7
#5217: 'Drone's Salvage Plasma Reworked into Salvage Biomass' by Surrealaser at commit 74d3442f2c
#5242: 'Reworks IFF code, Gives Aim Mode to select rifles' by Walarks at commit 259ba97f3d
#5253: 'Stuns changes part 2 - defender' by Pariah919 at commit 000ea3a015

## Reproduction:

[19:08:31] Runtime in dropship_ammo.dm, line 312: Cannot execute null.set up().
proc name: detonate on (/obj/structure/ship_ammo/minirocket/detonate_on)
usr: Iwan840/(Rei Nakano)
usr.loc: (Condor Jet (119, 83, 4))
src: the mini rocket stack (/obj/structure/ship_ammo/minirocket)
src.loc: the minirocket pod (/obj/structure/dropship_equipment/weapon/minirocket_pod)
call stack:
the mini rocket stack (/obj/structure/ship_ammo/minirocket): detonate on(the dirt floor (123,88,2) (/turf/open/floor/plating/ground/snow/layer0))
the minirocket pod (/obj/structure/dropship_equipment/weapon/minirocket_pod): open fire(Elder Praetorian (SPCR) (/mob/living/carbon/xenomorph/praetorian))

### PR DESCRIPTION
**Pull request by SplinterGP**
## About The Pull Request
Fixes some issues, for now this, will fix more runtimes.
Fixes #5365 
Fixes #5379 
Fixes #5377 
Fixes #5376 
Fixes #5380 

## Why It's Good For The Game
fixing issues.

## Changelog
:cl:
fix: fixes runtime errors caused by 0 lenght timers.
/:cl:

### CODE REVIEW

File: code/datums/components/riding.dm [hunk 1]
```diff
@@ -189,7 +189,7 @@
 		to_chat(user, "<span class='warning'>You'll need the keys in one of your hands to [drive_verb] [AM].</span>")
 
 /datum/component/riding/proc/Unbuckle(atom/movable/buckled_thing)
-	addtimer(CALLBACK(parent, /atom/movable/.proc/unbuckle_mob, buckled_thing), 0, TIMER_UNIQUE)
+	addtimer(CALLBACK(parent, /atom/movable/.proc/unbuckle_mob, buckled_thing), 1, TIMER_UNIQUE)
```
Comment (TiviPlus) [inline code/datums/components/riding.dm:192]: Why isnt this invoke async

Comment (SplinterGP) [reply code/datums/components/riding.dm:192]: doing now

**Review by Hughgent — APPROVED** [review]
Looking pretty good to me

**Comment by Hughgent** [comment]
another possible one in `aliens.dm` line 53 please check it out.

**Review by TiviPlus — APPROVED** [review]
This is good enough youdont haveto cram  them all into one PR

**Pull request merged**